tianshou/data/buffer/manager.py

```diff
@@ -118,20 +118,26 @@ def add(
     ) -> tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
         """Add a batch of data into ReplayBufferManager.
 
-        Each of the data's length (first dimension) must equal to the length of
-        buffer_ids. By default buffer_ids is [0, 1, ..., buffer_num - 1].
+        Each of the data's lengths (first dimension) must be equal to the length of
+        buffer_ids. By default, buffer_ids is [0, 1, ..., buffer_num - 1].
 
         Return (current_index, episode_reward, episode_length, episode_start_index). If
         the episode is not finished, the return value of episode_length and
         episode_reward is 0.
         """
+        # todo heavy code duplication with ReplayBuffer in buffer/base.py
         # preprocess batch
         new_batch = Batch()
         for key in set(self._reserved_keys).intersection(batch.keys()):
             new_batch.__dict__[key] = batch[key]
         batch = new_batch
         batch.__dict__["done"] = np.logical_or(batch.terminated, batch.truncated)
-        assert {"obs", "act", "rew", "terminated", "truncated", "done"}.issubset(batch.keys())
+        if not self._required_keys.issubset(batch.keys()):
```

Comment:
computing the difference and checking if it's empty is both faster and arguably clearer to read. You can even use the newer python syntax 

```python
if not missing_keys := self._required_keys.difference(batch.keys()):
   ...
```

Reply:
https://peps.python.org/pep-0572/

Reply:
nice stuff, thanks.